### PR TITLE
[zipper] add template

### LINKED ISSUE #2005: zipper: add test template
## Task
Create a test template file for `zipper`

### Details
1. If this is your first time contributing to this repository, welcome!
    - Contributing to a repository on GitHub requires a [fork](https://help.github.com/en/articles/fork-a-repo)
1. Create a branch on your local repository called `zipper-add-test-template`
    - Run: `git checkout -b zipper-add-test-template`
1. Clone problem-specifications repository; this is used to populate the template
    - Run: `git clone https://github.com/exercism/problem-specifications ../problem-specifications`
1. Create `exercises/zipper/.meta/template.j2`
    - See [Generator Documentation] for more information
1. Generate test file from new template
    - Run: `bin/generate_tests.py zipper`

At this point, if the generator fails, or you are otherwise stuck,
please feel free to reach out! It is perfectly acceptable to submit
an incomplete PR in order to get feedback or assistance.

### Resources
- [Contributing Guide]
- [Generator Documentation]
- [Closing issues using keywords]

[Contributing Guide]: https://github.com/exercism/python/blob/master/docs/CONTRIBUTING.md
[Generator Documentation]: https://github.com/exercism/python/blob/master/docs/GENERATOR.md
[Closing issues using keywords]: https://help.github.com/en/articles/closing-issues-using-keywords

### PR DESCRIPTION
**Pull request by yawpitch**
Adds a test template for `zipper`, removes the need for #2160, which was
going down a much more complicated path. Updates example.py for a minor
change in the canonical data for `Zipper.up()`.

Closes #2005.

### CODE REVIEW

File: exercises/zipper/.meta/template.j2 [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    {%- set expected = case["expected"] -%}
+    def test_{{ case["description"] | to_snake }}(self):
+        {%- if case["property"] == "sameResultFromOperations" %}
+        {{ same_result_from_operations(input, expected) }}
+        {%- else %}
+        {{ expected_value(input, expected) }}
+        {%- endif %}
+{%- endmacro -%}
+
+{%- macro expected_value(input, expected) %}
```
Comment (cmccandless) [inline exercises/zipper/.meta/template.j2]:
```suggestion
{%- macro expected_value(input, expected) -%}
```

File: exercises/zipper/.meta/template.j2 [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        {%- if expected["type"] == "tree" %}
+        self.assertEqual(result, expected) 
+        {%- elif expected["value"] is none %}
+        self.assertIsNone(result)
+        {%- else %}
+        self.assertEqual(result, {{ expected["value"] }})
+        {%- endif %}
+{%- endmacro %}
+
+{%- macro same_result_from_operations(input, expected) %}
```
Comment (cmccandless) [inline exercises/zipper/.meta/template.j2]:
```suggestion
{%- macro same_result_from_operations(input, expected) -%}
```

**Comment by yawpitch** [comment]
> Suggest trimming whitespace after macro start to remove that blank line after test function declaration.

Fixesd.

**Comment by cmccandless** [comment]
Hmm... looks like there's a problem with the `fetch-configlet` script. Wish we were printing out the URL we're attempting to fetch from...

**Comment by yawpitch** [comment]
> Hmm... looks like there's a problem with the `fetch-configlet` script. Wish we were printing out the URL we're attempting to fetch from...

I ran into this a couple days ago ... it seems to happen when there's some Github-side hiccup, because an empty commit a short time later fixed it.

But yeah I do wish we were printing the URL out.

**Pull request merged**